### PR TITLE
Delete WindowsSubsystemForLinux.md

### PR DESCRIPTION
**Pull request by compnerd**
This is duplicated by the Linux rules.  The Windows Subsystem for Linux is not a unique environment, it is an Ubuntu environment running with a bridge into the NT kernel.  There is nothing to be done to support it, just follow the Linux rules.  The Windows documentation has been updated to point users to the Linux section.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Comment by jrose-apple** [comment]
A bunch of these instructions differed from the Linux instructions. Should we have notes in the Linux instructions for "platforms other than Ubuntu"?

**Comment by compnerd** [comment]
@jrose-apple - ooh, I think that we should track those differences.  I bet that they exist because the two diverged.  The environment is exactly Ubuntu, so the build instructions should be identical (modulo things fixed here and not in the main bits, and vice versa).

File: docs/WindowsSubsystemForLinux.md [hunk 1]
```diff
@@ -1,60 +0,0 @@
...
-Install and run the latest version of [Bash on Ubuntu on
-Windows](https://docs.microsoft.com/en-us/windows/wsl/about) installed on
-your PC.
-```
-bash
-```
-
-### 2. Install dependencies
-Install the developer dependencies needed to compile the Swift project. These
-are identical to the Ubuntu dependencies, with the addition of `make`.
```
Comment (jrose-apple) [inline docs/WindowsSubsystemForLinux.md]: Here

File: docs/WindowsSubsystemForLinux.md [hunk 2]
```diff
@@ -1,60 +0,0 @@
...
-```bash
-sudo apt-get install git make cmake ninja-build clang python uuid-dev libicu-dev icu-devtools libbsd-dev libedit-dev libxml2-dev libsqlite3-dev swig libpython-dev libncurses5-dev pkg-config libblocksruntime-dev libcurl4-openssl-dev
-```
-
-### 3. Upgrade `clang`
-Install a version of `clang` with C++ 14 support; the default version of `clang`
-on WSL results in linker errors during compilation.
```
Comment (jrose-apple) [inline docs/WindowsSubsystemForLinux.md]: Here

File: docs/WindowsSubsystemForLinux.md [hunk 3]
```diff
@@ -1,60 +0,0 @@
...
-```bash
-sudo apt-get install clang-3.6
-sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-3.6 100
-sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-3.6 100
-```
-
-### 4. Upgrade CMake
-Install the latest version of CMake; Swift uses new CMake features such as
-`IN_LIST` and won't build without these features.
```
Comment (jrose-apple) [inline docs/WindowsSubsystemForLinux.md]: Here

**Review by jrose-apple — COMMENTED** [review]
I mean…

File: docs/WindowsSubsystemForLinux.md (hunk 1 above)
Comment (compnerd) [reply docs/WindowsSubsystemForLinux.md]: Ah.  I don't see why `make` is needed.  This is just wrong.

File: docs/WindowsSubsystemForLinux.md (hunk 2 above)
Comment (compnerd) [reply docs/WindowsSubsystemForLinux.md]: Hahah, I bet that is due to BFD rather than gold :-).

File: docs/WindowsSubsystemForLinux.md (hunk 3 above)
Comment (compnerd) [reply docs/WindowsSubsystemForLinux.md]: This is something that we would hit on any Ubuntu environment.  This I think is something that just diverged between the time that the docs were split.

File: docs/WindowsSubsystemForLinux.md (hunk 1 above)
Comment (jrose-apple) [reply docs/WindowsSubsystemForLinux.md]: Maybe to build `ninja` at the very start?

**Comment by compnerd** [comment]
Seems like the differences are due to incorrect documentation or diverge.  I'm going to merge this.

**Pull request merged**